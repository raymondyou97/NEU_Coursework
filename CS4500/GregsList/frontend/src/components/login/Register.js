'use es6';

import React, { PureComponent } from 'react';

import Button from 'react-bootstrap/Button';
import FormControl from 'react-bootstrap/FormControl';
import Form from 'react-bootstrap/Form';

import '../../styles/register.css';

class Register extends PureComponent {
  updateUsername = e => {
    this.props.updateUsername(e.target.value);
  };

  updateFirstName = e => {
    this.props.updateFirstName(e.target.value);
  };

  updateLastName = e => {
    this.props.updateLastName(e.target.value);
  };

  updatePassword = e => {
    this.props.updatePassword(e.target.value);
  };

  updateConfirmPassword = e => {
    this.props.updateConfirmPassword(e.target.value);
  };

  passwordsMatch = () => {
    return this.props.password === this.props.confirmPassword;
  };

  renderSubmit() {
    const {
      username,
      firstName,
      lastName,
      password,
      confirmPassword,
    } = this.props;

    const isButtonEnabled =
      username.length &&
      firstName.length &&
      lastName.length &&
      password.length &&
      confirmPassword.length &&
      this.passwordsMatch();

    return (
      <Button
        className="submit-button"
        disabled={!isButtonEnabled}
        onClick={this.props.createUser}
        variant="primary"
      >
        Register
      </Button>
    );
  }

  renderInputForms() {
    return (
      <Form>
        <Form.Label className="register-label">Username</Form.Label>
        <FormControl
          className="register-input username-input"
          onChange={this.updateUsername}
          placeholder="Username"
          value={this.props.username}
        />
        <Form.Label className="register-label">First name</Form.Label>
        <FormControl
          className="register-input firstname-input"
          onChange={this.updateFirstName}
          placeholder="First name"
          value={this.props.firstName}
        />
        <Form.Label className="register-label">Last name</Form.Label>
        <FormControl
          className="register-input lastname-input"
          onChange={this.updateLastName}
          placeholder="Last name"
          value={this.props.lastName}
        />
        <Form.Label className="register-label">Password</Form.Label>
        <FormControl
          className="register-input password-input"
          onChange={this.updatePassword}
          placeholder="Password"
          type="password"
          value={this.props.password}
        />
        <Form.Label className="register-label">Confirm password</Form.Label>
        <FormControl
          className={`register-input confirm-input ${
            this.passwordsMatch() ? '' : 'passwords-fail'
          }`}
          onChange={this.updateConfirmPassword}
          placeholder="Confirm password"
          type="password"
          value={this.props.confirmPassword}
        />
        {this.props.error ? (
          <Form.Label className="register-label-error">
            Username already exists
          </Form.Label>
        ) : null}
        {this.renderSubmit()}
      </Form>
    );
  }

  render() {
    return (
      <div className="register-container">
        <h1>Sign up</h1>
        {this.renderInputForms()}
      </div>
    );
  }
}

export default Register;
