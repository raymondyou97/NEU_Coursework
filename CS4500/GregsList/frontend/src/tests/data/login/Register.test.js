'use es6';

import React from 'react';
import Register from '../../../components/login/Register';
import renderer from 'react-test-renderer';

test('Empty snapshot renders', () => {
  const component = renderer.create(
    <Register
      firstName={''}
      lastName={''}
      username={''}
      password={''}
      confirmPassword={''}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updatePassword={() => {}}
      updateConfirmPassword={() => {}}
      createUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data renders', () => {
  const component = renderer.create(
    <Register
      firstName={'asdf'}
      lastName={'asdf'}
      username={'asdf'}
      password={'asdf'}
      confirmPassword={'asdf'}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updatePassword={() => {}}
      updateConfirmPassword={() => {}}
      createUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data renders', () => {
  const component = renderer.create(
    <Register
      firstName={'asdf'}
      lastName={'asdf'}
      username={'asdf'}
      password={'asdf'}
      confirmPassword={'asdf2'}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updatePassword={() => {}}
      updateConfirmPassword={() => {}}
      createUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with user data renders', () => {
  const component = renderer.create(
    <Register
      firstName={'asdf'}
      lastName={'asdf'}
      username={'asdf'}
      password={''}
      confirmPassword={''}
      updateUsername={() => {}}
      updateFirstName={() => {}}
      updateLastName={() => {}}
      updatePassword={() => {}}
      updateConfirmPassword={() => {}}
      createUser={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
