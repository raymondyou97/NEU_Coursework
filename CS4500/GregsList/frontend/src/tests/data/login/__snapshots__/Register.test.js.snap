// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty snapshot renders 1`] = `
<div
  className="register-container"
>
  <h1>
    Sign up
  </h1>
  <form
    className=""
  >
    <label
      className="register-label form-label"
    >
      Username
    </label>
    <input
      className="register-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value=""
    />
    <label
      className="register-label form-label"
    >
      First name
    </label>
    <input
      className="register-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value=""
    />
    <label
      className="register-label form-label"
    >
      Last name
    </label>
    <input
      className="register-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value=""
    />
    <label
      className="register-label form-label"
    >
      Password
    </label>
    <input
      className="register-input password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <label
      className="register-label form-label"
    >
      Confirm password
    </label>
    <input
      className="register-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data renders 1`] = `
<div
  className="register-container"
>
  <h1>
    Sign up
  </h1>
  <form
    className=""
  >
    <label
      className="register-label form-label"
    >
      Username
    </label>
    <input
      className="register-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      First name
    </label>
    <input
      className="register-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Last name
    </label>
    <input
      className="register-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Password
    </label>
    <input
      className="register-input password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Confirm password
    </label>
    <input
      className="register-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="asdf"
    />
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data renders 2`] = `
<div
  className="register-container"
>
  <h1>
    Sign up
  </h1>
  <form
    className=""
  >
    <label
      className="register-label form-label"
    >
      Username
    </label>
    <input
      className="register-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      First name
    </label>
    <input
      className="register-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Last name
    </label>
    <input
      className="register-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Password
    </label>
    <input
      className="register-input password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Confirm password
    </label>
    <input
      className="register-input confirm-input passwords-fail form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="asdf2"
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Snapshot with user data renders 3`] = `
<div
  className="register-container"
>
  <h1>
    Sign up
  </h1>
  <form
    className=""
  >
    <label
      className="register-label form-label"
    >
      Username
    </label>
    <input
      className="register-input username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      First name
    </label>
    <input
      className="register-input firstname-input form-control"
      onChange={[Function]}
      placeholder="First name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Last name
    </label>
    <input
      className="register-input lastname-input form-control"
      onChange={[Function]}
      placeholder="Last name"
      value="asdf"
    />
    <label
      className="register-label form-label"
    >
      Password
    </label>
    <input
      className="register-input password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <label
      className="register-label form-label"
    >
      Confirm password
    </label>
    <input
      className="register-input confirm-input  form-control"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Register
    </button>
  </form>
</div>
`;
